tests/std/tests/LWG2597_complex_branch_cut/test.cpp

```diff
@@ -0,0 +1,168 @@
+
+                check_norm(pow(CT{-x, +zero}, CT{+yr, -yi}), exp(CT{log(x), +pi} * CT{+yr, -yi}));
+                check_norm(pow(CT{-x, -zero}, CT{+yr, -yi}), exp(CT{log(x), -pi} * CT{+yr, -yi}));
+
+                check_norm(pow(CT{-x, +zero}, CT{-yr, +yi}), exp(CT{log(x), +pi} * CT{-yr, +yi}));
+                check_norm(pow(CT{-x, -zero}, CT{-yr, +yi}), exp(CT{log(x), -pi} * CT{-yr, +yi}));
+
+                check_norm(pow(CT{-x, +zero}, CT{-yr, -yi}), exp(CT{log(x), +pi} * CT{-yr, -yi}));
+                check_norm(pow(CT{-x, -zero}, CT{-yr, -yi}), exp(CT{log(x), -pi} * CT{-yr, -yi}));
+            }
+        }
+    }
+
+    // acos
+    for (const T x : {one, one_up, two, huge, inf}) {
+        check_part(acos(CT{+x, +zero}), CT{zero, -acosh(x)});
+        check_part(acos(CT{+x, -zero}), CT{zero, +acosh(x)});
+
+        check_part(acos(CT{-x, +zero}), CT{pi, -acosh(x)});
+        check_part(acos(CT{-x, -zero}), CT{pi, +acosh(x)});
+    }
+
+    // asin
+    for (const T x : {one, one_up, two, huge, inf}) {
+        check_part(asin(CT{+x, +zero}), CT{+pi_2, +acosh(x)});
+        check_part(asin(CT{+x, -zero}), CT{+pi_2, -acosh(x)});
+
+        check_part(asin(CT{-x, +zero}), CT{-pi_2, +acosh(x)});
+        check_part(asin(CT{-x, -zero}), CT{-pi_2, -acosh(x)});
+    }
+
+    // atan
+    for (const T x : {one_up, two, huge, inf}) {
+        check_part(atan(CT{+zero, +x}), CT{+pi_2, atanh(one / +x)});
+        check_part(atan(CT{-zero, +x}), CT{-pi_2, atanh(one / +x)});
+
+        check_part(atan(CT{+zero, -x}), CT{+pi_2, atanh(one / -x)});
+        check_part(atan(CT{-zero, -x}), CT{-pi_2, atanh(one / -x)});
+    }
+
+    // acosh
+    for (const T x : {one, one_down, half, tiny, zero}) {
+        check_part(acosh(CT{+x, +zero}), CT{zero, +acos(+x)});
+        check_part(acosh(CT{+x, -zero}), CT{zero, -acos(+x)});
+
+        check_part(acosh(CT{-x, +zero}), CT{zero, +acos(-x)});
+        check_part(acosh(CT{-x, -zero}), CT{zero, -acos(-x)});
+    }
+
+    for (const T x : {one, one_up, two, huge, inf}) {
+        check_part(acosh(CT{-x, +zero}), CT{acosh(x), +pi});
+        check_part(acosh(CT{-x, -zero}), CT{acosh(x), -pi});
+    }
+
+    // asinh
+    for (const T x : {one, one_up, two, huge, inf}) {
+        check_part(asinh(CT{+zero, +x}), CT{+acosh(x), +pi_2});
+        check_part(asinh(CT{-zero, +x}), CT{-acosh(x), +pi_2});
+
+        check_part(asinh(CT{+zero, -x}), CT{+acosh(x), -pi_2});
+        check_part(asinh(CT{-zero, -x}), CT{-acosh(x), -pi_2});
+    }
+
+    // atanh
+    for (const T x : {one_up, two, huge, inf}) {
+        check_part(atanh(CT{+x, +zero}), CT{atanh(one / +x), +pi_2});
+        check_part(atanh(CT{+x, -zero}), CT{atanh(one / +x), -pi_2});
+
+        check_part(atanh(CT{-x, +zero}), CT{atanh(one / -x), +pi_2});
+        check_part(atanh(CT{-x, -zero}), CT{atanh(one / -x), -pi_2});
+    }
+}
+
+int main() {
+    test<double>();
+    test<float>();
+    test<long double>();

```

Comment:
Maybe test for generic implementation as well?

Reply:
Or is it broken due to #519 ? Maybe comment if so?

Reply:
The test is only valid for floating point types with signed zeros.

`complex<T>` for `T` other than `float`, `double` or `long double` is unspecified in the standard, and MSVC STL possibly supports `complex<Integral>` to a certain degree (see #756), but I don't believe that elementary functions are supported (e.g. `sqrt(complex<int>{1, 0})` crashes).

Reply:
I propose to deprecate the nonstandard extension, if it cannot be made working correctly. See #889